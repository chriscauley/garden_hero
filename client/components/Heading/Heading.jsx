import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { buildClassName } from 'utils/util';
import { headingProps, statusProps } from 'utils/propTypeEnum';

class Heading extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    element: PropTypes.string,
    modifier: PropTypes.oneOf(headingProps.modifiers),
    status: PropTypes.oneOf(statusProps.types),
    section: PropTypes.bool,
    size: PropTypes.oneOf(headingProps.sizes)
  };

  static defaultProps = {
    element: 'span',
    size: 'l',
    section: false
  };

  render() {
    const {
      children,
      className,
      element: Element,
      modifier,
      size,
      status,
      section,
      ...props
    } = this.props;
    const mainClass = 'bcp-heading';
    const modifierClass = buildClassName(
      headingProps.modifiers,
      modifier,
      mainClass
    );
    const sizeClass = buildClassName(headingProps.sizes, size, mainClass);
    const statusClass = buildClassName(statusProps.types, status, mainClass);
    const sectionClass = section
      ? buildClassName(['section'], 'section', mainClass)
      : null;

    return (
      <Element
        className={classNames(
          mainClass,
          modifierClass,
          sizeClass,
          statusClass,
          sectionClass,
          className
        )}
        {...props}
      >
        {children}
        {section ? <span className="divider" /> : null}
      </Element>
    );
  }
}

export default Heading;
