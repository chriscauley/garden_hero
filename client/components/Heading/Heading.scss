.bcp-heading {
    display: block;
    margin-bottom: em(10px);
    color: $grayDark;
    &--inline {
        display: inline;
    }
    &--xl {
        @include heading-xl;
        font-weight: $fontWeightLight;
    }
    &--l {
        @include heading-l;
        font-weight: $fontWeightRegular;
    }
    &--s {
        @include heading-s;
        font-weight: $fontWeightRegular;
    }
    &--bold {
        font-weight: $fontWeightBold;
    }
    &--success {
        color: $green;
    }
    &--error {
        color: $red;
    }
    &--section {
        display: flex;
        align-items: center;
        margin-bottom: em(30px);
        line-height: 1;
        >.bcp-icon {
            fill: $grayDark;
            margin-right: 0.5em;
        }
        .divider {
            flex: 2;
            border-top: 1px solid $grayMedium;
            margin-left: 20px;
            height: 1px;
        }
    }
}