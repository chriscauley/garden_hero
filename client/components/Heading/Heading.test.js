import React from 'react';
import renderer from 'react-test-renderer';

import Heading from './Heading';

describe('<Heading />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Heading>Welcome</Heading>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders child elements', () => {
    const component = renderer.create(
      <Heading>
        Welcome <i />
      </Heading>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('accepts className prop', () => {
    const component = renderer.create(
      <Heading className="gh-margin-top">Welcome</Heading>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('accepts modifier prop', () => {
    const component = renderer.create(
      <Heading modifier="inline">Welcome</Heading>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('accepts size prop', () => {
    const component = renderer.create(<Heading size="l">Welcome</Heading>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('accepts status prop', () => {
    const component = renderer.create(
      <Heading status="success">Welcome</Heading>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders section heading', () => {
    const component = renderer.create(
      <Heading size="l" section>
        Welcome
      </Heading>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
