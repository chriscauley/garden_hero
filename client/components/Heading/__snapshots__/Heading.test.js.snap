// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> accepts className prop 1`] = `
<span
  className="gh-heading gh-heading--l gh-margin-top"
>
  Welcome
</span>
`;

exports[`<Heading /> accepts modifier prop 1`] = `
<span
  className="gh-heading gh-heading--inline gh-heading--l"
>
  Welcome
</span>
`;

exports[`<Heading /> accepts size prop 1`] = `
<span
  className="gh-heading gh-heading--l"
>
  Welcome
</span>
`;

exports[`<Heading /> accepts status prop 1`] = `
<span
  className="gh-heading gh-heading--l gh-heading--success"
>
  Welcome
</span>
`;

exports[`<Heading /> renders child elements 1`] = `
<span
  className="gh-heading gh-heading--l"
>
  Welcome 
  <i />
</span>
`;

exports[`<Heading /> renders correctly 1`] = `
<span
  className="gh-heading gh-heading--l"
>
  Welcome
</span>
`;

exports[`<Heading /> renders section heading 1`] = `
<span
  className="gh-heading gh-heading--l gh-heading--section"
>
  Welcome
  <span
    className="divider"
  />
</span>
`;
